compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1450,6 +1450,13 @@ object SymDenotations {
       else if is(Contravariant) then Contravariant
       else EmptyFlags
 
+    /** The length of the owner chain of this symbol. 0 for NoSymbol, 1 for _root_ */
+    def nestingLevel(using Context): Int =
+      @tailrec def recur(d: SymDenotation, n: Int): Int = d match
+        case d: ClassDenotation => d.nestingLevel + n // profit from the cache in ClassDenotation
+        case _ => recur(d.owner, n + 1)
```

Comment:
I thought it was preferable to change the comment instead. So, `nestingLevel` of _root_ is 0, and it's undefined for NoSymbol. We don't need it for NoSymbol, and the code is simpler that way.

Reply:
>  We don't need it for NoSymbol

In TypeVar#avoidCaptures we call `ref.symbol.maybeOwner.nestingLevel`, I think this will crash if ref happens to be a reference to a member of a structural type.

Reply:
Also, ClassDenotation#nestingLevel does check if the owner exists, is that difference intentional?

Reply:
Thinking about it more, the case of structural type member is interesting: it seems that their level should be the level of their underlying type, but then that means `nestingLevel` should be defined on `Type`

Reply:
Currently, TypeVar#avoidCaptures only look for TermRef, and I think these TermRef can never be structural members because a reference to such a member is never a stable path. So I think the current implementation is OK, but it's something to watch out for if Typevar#avoidCaptures is extended to do more checks

Reply:
I fixed `nestingLevel` to work for `NoDenotatoin` as well. Structural type members would not have the form `TermRef(NoPrefix, _)`, so we do not check their nestingLevel. It's true that we might also have to extend this to type members. That's part of #8900.